### PR TITLE
[Common] Logical fix in machine ops

### PR DESCRIPTION
**Pull request by srijan-sivakumar**
The patch fixes the logic in
1. Snap lv remove function
2. Convert hostname to IP

Fixes: #892

Signed-off-by: srijan-sivakumar <ssivakum@redhat.com>

### CODE REVIEW

**Comment by srijan-sivakumar** [comment]
```console
(redant-demo) [root@rhsdev-virt05 redant]# python3 core/redant_main.py -c ../rconfig.yml -t tests/example/sample_component/test_glusterd_ops.py -ll D
    ____  __________  ___    _   ________
   / __ \/ ____/ __ \/   |  / | / /_  __/
  / /_/ / __/ / / / / /| | /  |/ / / /   
 / _, _/ /___/ /_/ / ___ |/ /|  / / /    
/_/ |_/_____/_____/_/  |_/_/ |_/ /_/     
                                         

✔ Param Handling Success.
✔ Test List built
✔ Log dir creation successful.
✔ Environment setup successful.
✔ Running test case : vol_create_test-dist
✔ vol_create_test-dist Succeeded
✔ Running test case : test_glusterd_ops-dist
✔ test_glusterd_ops-dist Succeeded
✔ Running test case : vol_destroy_test-dist
✔ vol_destroy_test-dist Succeeded
Results for Special
Test nature : nonDisruptive
TestName : vol_create_test
+-------------+-------------+-----------------------+-------------+
| Volume Type | Test Result | Time Taken (hh:mm:ss) | Skip Reason |
+-------------+-------------+-----------------------+-------------+
|     dist    |     PASS    |        00:00:07       |      NA     |
+-------------+-------------+-----------------------+-------------+
TestName : vol_destroy_test
+-------------+-------------+-----------------------+-------------+
| Volume Type | Test Result | Time Taken (hh:mm:ss) | Skip Reason |
+-------------+-------------+-----------------------+-------------+
|     dist    |     PASS    |        00:00:09       |      NA     |
+-------------+-------------+-----------------------+-------------+
Results for sample_component
Test nature : nonDisruptive
TestName : test_glusterd_ops
+-------------+-------------+-----------------------+-------------+
| Volume Type | Test Result | Time Taken (hh:mm:ss) | Skip Reason |
+-------------+-------------+-----------------------+-------------+
|     dist    |     PASS    |        00:00:29       |      NA     |
+-------------+-------------+-----------------------+-------------+
+--------------------------------------+-------+
|               Category               | Count |
+--------------------------------------+-------+
|           Disruptive Tests           |   0   |
|         Non Disruptive Tests         |   3   |
|       Disruptive Tests Skipped       |   0   |
|     Non Disruptive Tests Skipped     |   0   |
|              Tests Ran               |   3   |
|            Tests Skipped             |   0   |
|             Total Tests              |   3   |
|   Disruptive Tests Pass Percentage   |   0   |
| Non Disruptive Tests Pass Percentage | 100.0 |
|        Total Pass Percentage         | 100.0 |
+--------------------------------------+-------+
Total Time Taken : 00 hours 01 minutes 01 seconds
✔ Tearing down successful.
```

File: common/ops/support_ops/machine_ops.py [hunk 1]
```diff
@@ -290,13 +290,15 @@ def convert_hosts_to_ip(self, node_list: list, node: str = None) -> list:
 
         # Replace localhost with its hostname
         if 'localhost' in node_list:
+            new_node_list.append(node)
```
Comment (nik-redhat) [inline common/ops/support_ops/machine_ops.py:293]: Appending here as well, wouldn't make the list have the same value twice?

Comment (nik-redhat) [reply common/ops/support_ops/machine_ops.py:293]: Oh, I see we the node is no more present in the actual list to occur twice.

Comment (nik-redhat) [reply common/ops/support_ops/machine_ops.py:293]: But again, if we append it directly then we might get the host with a FQDN happened directly which won't be converted

Comment (nik-redhat) [reply common/ops/support_ops/machine_ops.py:293]: I think this shouldn't be here, and the simple remove and append as previous are fine. Let me know if I am thinking in the wrong direction...

Comment (srijan-sivakumar) [reply common/ops/support_ops/machine_ops.py:293]: But the `node` is actually passed by us and that is nothing but an ip value. basically this is the node wherein we tell rexe to run the commands.

Comment (srijan-sivakumar) [reply common/ops/support_ops/machine_ops.py:293]: on running with a debugger, this is what I got,
node_list ( before the localhost removal ) : ['10.xx.yy.aa', '10.xx.yy.bb', 'localhost']
new_node_list : ['10.xx.yy.cc', '10.xx.yy.bb', '10.xx.yy.aa']

and '10.xx.yy.xx' is the value of node.

Comment (nik-redhat) [reply common/ops/support_ops/machine_ops.py:293]: Got it...

**Review by nik-redhat — APPROVED** [review]

**Pull request merged**